lib/state/action-types.ts

```diff
@@ -1,3 +1,4 @@
 export const AUTH_SET = 'AUTH_SET';
 export const FILTER_NOTES = 'FILTER_NOTES';
+export const SET_SELECTED_NOTE = 'SET_SELECTED_NOTE';
```

Comment:
as you work here please be thinking about how best to represent our Redux store, actions, and selectors.

since we have a very minimal store to start with _and_ we're 100% TypeScript there are ways we can think about changing this to be more helpful in the IDE. if you haven't had the chance you can play around with some of the ideas in #1820. some things in there I like and others I don't.

Reply:
What do you think of this?  I still don't have my brain fully wrapped on TS.

https://redux.js.org/recipes/usage-with-typescript/#type-checking-actions--action-creators

Reply:
We'll discuss it beyond this PR, but an earlier incantation of #1820 looked closer to your link. I changed it to try and minimize the number of named imports and improve the auto-complete while writing reducers and `connect()` functions.

Although I've grown accustomed to inferring the types of `mapStateToProps` I have come to like separating it into separate `OwnProps`, `DispatchProps`, and `StateProps` because of how they ease the typing of `mapStateToProps`, `mapDispatchToProps`, and `connect()`.

Reply:
after the merge of #1855 and #1856 we didn't remove the existing constants but we don' need to add new ones here either. I'll try and get those existing ones removed in a separate PR so they don't stick around and confuse us.